pkg/webui/oauth/views/app/index.js

```diff
@@ -78,7 +80,9 @@ export default class OAuthApp extends React.PureComponent {
                     <Route path="/oauth/login" component={Login} />
                     <Route path="/oauth/authorize" component={Authorize} />
                     <Route path="/oauth/register" component={CreateAccount} />
+                    <Route path="/oauth/reset-password" component={ResetPassword} />
                     <Route path="/oauth/code" component={Code} />
+                    <Route path="/oauth/change-password/:user_id/:token" component={ChangePassword} />
```

Comment:
It's not `token`. It's `current_password` or in some cases `temporary_password`. Also I wouldn't put that in the path but in the query.

Reply:
I think temporary password is a very confusing naming, since you cannot log in with it. The definition of a password is:
> a secret word or phrase that must be used to gain admission to a place.
IMO, this is a password reset token, only used when changing password. But if it's too much work to change the naming for the sake of clarity, I am fine with it as well.

Reply:
Also, not sure what you mean by query. Would we query user id that is in the input for the temporary password associated for them and then make an api request for changing password? Wouldn't that mean that anyone could reset and change a password for any user?

Reply:
The term "temporary password" has been part of our API since October last year, it's a little bit too late to change it now.

What I mean with query is that we could put a link in the password reset email that prefills form fields of the "change password" form, for example:

http://localhost:1885/oauth/change-password?user_id=admin&current=supersecret

which would prefill the `user_id` and `current` fields with the given contents.

Reply:
I don't think we need to show the `user_id` and `temporary_password` fields prefilled in the reset form anyway. Both should be redundant for what the user is trying to do, he/she just cares about changing the password. This is also the flow I know from other apps.

Just two fields for changing the password should suffice, the request body can then be extended with the info from the query.

Implementing two slightly different views for the two scenarios of resetting after forgetting (2 fields) and simply changing the password (3 fields) is not expensive IMO.

@htdvisser would you be ok with that?